# Análise Exploratória de Dados

A partir do dataset formatado na etapa anterior, será feita uma análise exploratória dos dados

In [1]:
import pandas as pd

In [2]:
df = pd.read_json('./data_files/merged_datasets.json', orient='split')
df

Text    Gene  \
0     Cyclin-dependent kinases (CDKs) regulate a var...  FAM58A   
1      Abstract Background  Non-small cell lung canc...     CBL   
2      Abstract Background  Non-small cell lung canc...     CBL   
3     Recent evidence has demonstrated that acquired...     CBL   
4     Oncogenic mutations in the monomeric Casitas B...     CBL   
...                                                 ...     ...   
3316  Introduction  Myelodysplastic syndromes (MDS) ...   RUNX1   
3317  Introduction  Myelodysplastic syndromes (MDS) ...   RUNX1   
3318  The Runt-related transcription factor 1 gene (...   RUNX1   
3319  The RUNX1/AML1 gene is the most frequent targe...   RUNX1   
3320  The most frequent mutations associated with le...   RUNX1   

                 Variation  Class  
0     Truncating Mutations      1  
1                    W802*      2  
2                    Q249E      2  
3                    N454D      3  
4                    L399V      4  
...                    ...    ...  
3316                 D171N      4  
3317                 A122*      1  
3318               Fusions      1  
3319                  R80C      4  
3320                  K83E      4  

[3321 rows x 4 columns]

In [6]:
# Verificando valores NA's
df.isna().sum()

Text         0
Gene         0
Variation    0
Class        0
dtype: int64

In [8]:
# Verificando os tipos das variáveis
df.dtypes

Text         object
Gene         object
Variation    object
Class         int64
dtype: object

In [11]:
# Verificando valores únicos
df.nunique()

Text         1921
Gene          264
Variation    2996
Class           9
dtype: int64

Notar que enquanto o dataset possui 3221 observações, há apenas 1921 textos distintos e 2996 valores de Variation distintos. Isto significa que pode haver mais de uma variação para cada texto descritivo.

In [13]:
# Verificando se há linhas duplicadas
df.duplicated().sum()

0

Através de inspeção no arquivo, o atributo Variation traz o nome da variação, com exceção de algumas classes comuns como deleção, fusão, etc. Deste modo, conclui-se que não acrescenta informação relevante à análise e será excluída do dataset.

In [15]:
df.drop('Variation', inplace=True, axis=1)

In [16]:
df

Text    Gene  Class
0     Cyclin-dependent kinases (CDKs) regulate a var...  FAM58A      1
1      Abstract Background  Non-small cell lung canc...     CBL      2
2      Abstract Background  Non-small cell lung canc...     CBL      2
3     Recent evidence has demonstrated that acquired...     CBL      3
4     Oncogenic mutations in the monomeric Casitas B...     CBL      4
...                                                 ...     ...    ...
3316  Introduction  Myelodysplastic syndromes (MDS) ...   RUNX1      4
3317  Introduction  Myelodysplastic syndromes (MDS) ...   RUNX1      1
3318  The Runt-related transcription factor 1 gene (...   RUNX1      1
3319  The RUNX1/AML1 gene is the most frequent targe...   RUNX1      4
3320  The most frequent mutations associated with le...   RUNX1      4

[3321 rows x 3 columns]

In [17]:
# Salvando o resultado intermediário em um json
df.to_json('./data_files/df_text_gene_class.json')